sys/include/net/sock/tcp.h

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (C) 2016 Alexander Aring <aar@pengutronix.de>
+ *                    Freie Universität Berlin
+ *                    HAW Hamburg
+ *                    Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_sock_tcp    TCP sock API
+ * @ingroup     net_sock
+ * @brief       Sock submodule for TCP
+ * @{
+ *
+ * @file
+ * @brief   TCP sock definitions
+ *
+ * @author  Alexander Aring <aar@pengutronix.de>
+ * @author  Simon Brummer <simon.brummer@haw-hamburg.de>
+ * @author  Cenk Gündoğan <mail@cgundogan.de>
+ * @author  Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ * @author  Kaspar Schleiser <kaspar@schleiser.de>
+ */
+#ifndef NET_SOCK_TCP_H_
+#define NET_SOCK_TCP_H_
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "net/sock/addr.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   An end point for a TCP sock object
+ */
+typedef struct {
+    sock_addr_ip_t addr;    /**< IP address */
+    int family;             /**< family of sock_tcp_ep_t::addr */
```

Comment:
At least on 32-bit platforms those would return anyways due to alignment. But I'm thinking of moving it to the top, that so that endpoints can be identified to a degree even after casting, if `addr` isn't an IP address, but e.g. a name for an ICN-based `sock`. And then `int` is the only type that would make sense, due to the alignment of the following array.